### DISCUSSION TITLE
React 16 & 17 support

### DISCUSSION THREAD

**Discussion by tmeindle** [TanStack/router#751] [Q&A]
https://github.com/TanStack/router/blob/0e26a516c14a0eef7e7d3c40b86efba87831a3d4/packages/react-router/package.json#L41-L42

Are the peer dependencies correct?  Is this library supposed to support react < 18?

The RouterProvider uses a startTransition with out any way to skip the call which generates a runtime error when used with react 17.0.2 (which I am currently restricted too)

https://github.com/TanStack/router/blob/0e26a516c14a0eef7e7d3c40b86efba87831a3d4/packages/react-router/src/react.tsx#L637

Is the react transition api a requirement of the router working correctly?  Could there be a router provider prop that would default links and router provider to not use the startTransition calls.

**Comment by tmeindle** [comment]
I tried removing the calls to startTransition and while the router works it seems to re-render the entire app for every route change (i.e. the window blinks).  No dialog after 3 weeks, I am going to assume that react < 18 is not going to be supported.

**Comment by jdefting** [comment]
I'm wondering this to. Release v1 is out and the docs say the library supports v16.8+. 

https://github.com/TanStack/router/blob/912fae7e67f8a5b6a9b7445044393e3259a2a717/docs/installation.md?plain=1#L11

But there's still `useTransition()` usage which is React 18 API.

https://github.com/TanStack/router/blob/912fae7e67f8a5b6a9b7445044393e3259a2a717/packages/react-router/src/RouterProvider.tsx#L110

**Comment by jdefting** [comment]
There shouldn't be issues with useTransition now: https://github.com/TanStack/router/issues/900
